module/x/peggy/keeper/test_common.go

```diff
@@ -21,23 +23,149 @@ import (
 	dbm "github.com/tendermint/tm-db"
 )
 
-func CreateTestEnv(t *testing.T) (Keeper, sdk.Context) {
+var TestingStakeParams = staking.Params{
+	UnbondingTime:     100,
+	MaxValidators:     10,
+	MaxEntries:        10,
+	HistoricalEntries: 10,
+	BondDenom:         "stake",
+}
+
+type TestKeepers struct {
+	AccountKeeper auth.AccountKeeper
+	StakingKeeper staking.Keeper
+	DistKeeper    distribution.Keeper
+	SupplyKeeper  supply.Keeper
+	GovKeeper     gov.Keeper
+	BankKeeper    bank.Keeper
+}
+
+func CreateTestEnv(t *testing.T) (Keeper, sdk.Context, TestKeepers) {
 	t.Helper()
 	peggyKey := sdk.NewKVStoreKey(types.StoreKey)
+	keyAcc := sdk.NewKVStoreKey(auth.StoreKey)
+	keyStaking := sdk.NewKVStoreKey(staking.StoreKey)
+	keySupply := sdk.NewKVStoreKey(supply.StoreKey)
+	keyDistro := sdk.NewKVStoreKey(distribution.StoreKey)
+	keyParams := sdk.NewKVStoreKey(params.StoreKey)
+	tkeyParams := sdk.NewTransientStoreKey(params.TStoreKey)
+	keyGov := sdk.NewKVStoreKey(govtypes.StoreKey)
 
 	db := dbm.NewMemDB()
 	ms := store.NewCommitMultiStore(db)
 	ms.MountStoreWithDB(peggyKey, sdk.StoreTypeIAVL, db)
+	ms.MountStoreWithDB(keyAcc, sdk.StoreTypeIAVL, db)
+	ms.MountStoreWithDB(keyParams, sdk.StoreTypeIAVL, db)
+	ms.MountStoreWithDB(keyStaking, sdk.StoreTypeIAVL, db)
+	ms.MountStoreWithDB(keySupply, sdk.StoreTypeIAVL, db)
+	ms.MountStoreWithDB(keyDistro, sdk.StoreTypeIAVL, db)
+	ms.MountStoreWithDB(tkeyParams, sdk.StoreTypeTransient, db)
+	ms.MountStoreWithDB(keyGov, sdk.StoreTypeIAVL, db)
 	err := ms.LoadLatestVersion()
 	require.Nil(t, err)
 
+	const isCheckTx = false
 	ctx := sdk.NewContext(ms, abci.Header{
 		Height: 1234567,
 		Time:   time.Date(2020, time.April, 22, 12, 0, 0, 0, time.UTC),
-	}, false, log.NewNopLogger())
+	}, isCheckTx, log.NewNopLogger())
 
 	cdc := MakeTestCodec()
-	return NewKeeper(cdc, peggyKey, AlwaysPanicStakingMock{}), ctx
+
+	paramsKeeper := params.NewKeeper(cdc, keyParams, tkeyParams)
+
+	accountKeeper := auth.NewAccountKeeper(
+		cdc,    // amino codec
+		keyAcc, // target store
+		paramsKeeper.Subspace(auth.DefaultParamspace),
+		auth.ProtoBaseAccount, // prototype
+	)
+
+	// this is also used to initialize module accounts (so nil is meaningful here)
```

Comment:
expand on what nil means in this context.

Reply:
ja, this was copied from wasmd but I will l rephrase this to `// this is also used to initialize module accounts for all the map keys`. It is a big method to setup the system under tests. Not very readable...